js/gdm/realmd.js

```diff
@@ -127,6 +127,9 @@ var Manager = new Lang.Class({
             }
         }
 
+        if (!this._loginFormat && !newLoginFormat)
+            return;
```

Comment:
This early return will be taken if both `this._loginFormat` and `newLoginFormat` evaluate to `false`. In that case, we most likely have `this._loginFormat == newLoginFormat`, right? In which case, the branch below shouldn’t be taken and the function should exit without doing anything else anyway. Is that not the case?

Reply:
Now I look at this patch, I realize is completely wrong, since the problem I'm trying to avoid is a reference to a non-initialized `this._loginFormat` attribute, which would throw an error. But that can be achieved by properly initializing that attribute in the constructor, so I think I will do that instead.